<Window
    x:Class="VainZero.SandBox.Wpf.MainWindow"
    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
    xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
    xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
    xmlns:local="clr-namespace:VainZero.SandBox.Wpf"
    mc:Ignorable="d"
    Title="MainWindow"
    Width="525"
    Height="350"
    >
    <DockPanel>
        <Button x:Name="successButton" DockPanel.Dock="Top" Content="Success" Click="successButton_Click" />
        <Button x:Name="failureButton" DockPanel.Dock="Top" Content="Failure" Click="failureButton_Click" />

        <local:TaskControl Task="{Binding Value}">
            <local:TaskControl.ErrorTemplate>
                <DataTemplate>
                    <Border BorderThickness="1" BorderBrush="Red">
                        <TextBlock
                            HorizontalAlignment="Center"
                            VerticalAlignment="Center"
                            Text="{Binding InnerException.Message}"
                            FontWeight="Bold"
                            Foreground="Red"
                            />
                    </Border>
                </DataTemplate>
            </local:TaskControl.ErrorTemplate>

            <ListBox ItemsSource="{Binding}" />
        </local:TaskControl>
    </DockPanel>
</Window>
